### PR TITLE
fix fsdp device mesh depenency issue

### PR DESCRIPTION
**Pull request by wanchaol**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #121061

as reported in https://github.com/pytorch/torchtrain/pull/103

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang

### CODE REVIEW

File: torch/distributed/fsdp/_common_utils.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 import torch.distributed.fsdp._flat_param as flat_param_file
 import torch.nn as nn
 from torch.distributed._composable_state import _get_module_state, _State
-from torch.distributed._tensor.device_mesh import DeviceMesh
+from torch.distributed.device_mesh import DeviceMesh
```
Comment (awgu) [inline torch/distributed/fsdp/_common_utils.py:31]: The old import should still be valid 🤔 : https://github.com/pytorch/pytorch/blob/f7a2bae0ac4f4c3b594619c1e39867695501735d/torch/distributed/_tensor/__init__.py#L12

**Review by awgu — APPROVED** [review]
Stamp to unblock

**Comment by awgu** [comment]
Check lint before landing if choosing to land

File: torch/distributed/fsdp/_common_utils.py (hunk 1 above)
Comment (wanchaol) [reply torch/distributed/fsdp/_common_utils.py:31]: Yeah it's quite weird, this poping up some recursively imports issue in some CI and that's the motivation for the fix https://github.com/pytorch/torchtrain/actions/runs/8118646813/job/22193278282?pr=103

**Pull request closed without merging**